prime-router/src/test/kotlin/MapperTests.kt

```diff
@@ -499,7 +499,51 @@ class MapperTests {
 
     @Test
     fun `test parseMapperField validation - allow mapper tokens to be parsed`() {
-        val vals = Mappers.parseMapperField("concat(patient_id, \$index)")
+        // it should allow mapper tokens to be parsed: i.e. "$index"
+        var vals = Mappers.parseMapperField("concat(patient_id, \$index)")
         assertThat(vals.second[1]).isEqualTo("\$index")
+
+        // it should allow mapper tokens to be parsed with semi-colon literal values: i.e. "$dateFormat:some-valid-date-format"
+        vals = Mappers.parseMapperField("nullDateValidator(\$dateFormat:some-valid-date-format, test_result_date)")
+        assertThat(vals.second[0]).isEqualTo("\$dateFormat:some-valid-date-format")
+    }
+
+    @Test
+    fun `test NullDateValidator`() {
+        val mapper = NullDateValidator()
+        val elementA = Element("a")
+        val elementB = Element("b")
+
+        // $dateFormat:yyyyMMdd, a (element name)
+        // should return the original element's value
+        val args = listOf("yyyyMMdd", "a")
+        var value = listOf(ElementAndValue(elementA, "yyyyMMdd"), ElementAndValue(elementB, "20211028"))
+        assertThat(mapper.apply(elementA, args, value))
+            .isEqualTo("20211028")
+
+        // $dateFormat:MM/dd/yyyy, a (element name)
+        // should return the original element's value
+        val args2 = listOf("MM/dd/yyyy", "a")
+        value = listOf(ElementAndValue(elementA, "MM/dd/yyyy"), ElementAndValue(elementB, "10/28/2021"))
+        assertThat(mapper.apply(elementA, args2, value))
+            .isEqualTo("10/28/2021")
+
+        // mismatched formatting
+        // should return an empty string
+        value = listOf(ElementAndValue(elementA, "yyyyMMdd"), ElementAndValue(elementB, "a week ago"))
+        assertThat(mapper.apply(elementA, args, value))
+            .isEqualTo("")
+
+        // empty values
+        // should return an empty string
+        value = listOf()
+        assertThat(mapper.apply(elementA, args, value))
+            .isEqualTo("")
+
+        // empty args
+        // should return an empty string
+        value = listOf(ElementAndValue(elementA, "yyyyMMdd"), ElementAndValue(elementB, "a week ago"))
```

Comment:
Add some error cases like pattern is invalid.

Reply:
@carlosfelix2 not exactly sure what you mean? we're already doing that above with the "a week ago" case, for example...

Reply:
how about listOf(ElementAndValue(elementA, "xxxxxx"), ElementAndValue(elementB, "a week ago"))
Or some other pattern that is invalid?  Try to break your code.

Reply:
@carlosfelix2 think I finally got you. added tests